Packs/Slack/Integrations/SlackV3/SlackV3.py

```diff
@@ -119,62 +122,209 @@ def get_current_utc_time() -> datetime:
+    while True:
+        workspace_users = response['members'] if response and response.get('members',
+                                                                           []) else []
+        cursor = response.get('response_metadata', {}).get('next_cursor')
+        user = return_user_filter(user_to_search.lower(), workspace_users)
+        if user:
+            break
+        if not cursor:
+            break
+        body = body.copy()  # strictly for unit testing purposes
+        body.update({'cursor': cursor})
         response = send_slack_request_sync(CLIENT, 'users.list', http_verb='GET', body=body)
-        while True:
-            workspace_users = response['members'] if response and response.get('members',
-                                                                               []) else []
-            cursor = response.get('response_metadata', {}).get('next_cursor')
-            users_filter = list(filter(lambda u: u.get('name', '').lower() == user_to_search
-                                       or u.get('profile', {}).get('email', '').lower() == user_to_search
-                                       or u.get('real_name', '').lower() == user_to_search, workspace_users))
-            if users_filter:
-                break
-            if not cursor:
-                break
-            body = body.copy()  # strictly for unit testing purposes
-            body.update({'cursor': cursor})
-            response = send_slack_request_sync(CLIENT, 'users.list', http_verb='GET', body=body)
-
-        if users_filter:
-            user = users_filter[0]
-            if add_to_context:
-                users.append(user)
-                set_to_integration_context_with_retries({'users': users}, OBJECTS_TO_KEYS, SYNC_CONTEXT)
-        else:
-            return {}
 
+    if user:
+        demisto.debug(f"User {user_to_search} was found.")
+        return format_user_results(user)
+    else:
+        demisto.debug(format_user_not_found_error(user=user_to_search))
+        return {}
+
+
+def format_user_results(user: dict):
+    """
+    Formats and truncates the user result from the Slack API
+
+    Args:
+        user: The user response retrieved from the Slack API
+    Returns:
+        Formatted and truncated version of the result which is context safe.
+    """
+    return {
+        'name': user.get('name'),
+        'id': user.get('id'),
+        'profile': {
+            'email': user.get('profile', {}).get('email', ''),
+            'real_name': user.get('profile', {}).get('real_name', ''),
+            'display_name': user.get('profile', {}).get('display_name', ''),
+        }
+    }
+
+
+def get_user_by_name_caching_disabled(user_to_search: str) -> dict:
+    """
+    When Disable Caching is true, we look for the user by email only.
+
+    Args:
+        user_to_search: The user's email
+    Returns:
+        A slack user object
+    """
+    user = {}
+    if re.match(emailRegex, user_to_search):
```

Comment:
is this needed if you do the same check in get_user_by_email()?

Reply:
Yes because users can still use something other than an email and we want to catch it.